src/main/kotlin/test/ktortemplate/Application.kt

```diff
@@ -43,9 +46,17 @@ fun Application.module(configOverrides: ApplicationConfig? = null) {
         }
     }
 
+    // Installs call logging and request tracing
+    val callIdHeader = environment.config.propertyOrNull("ktor.callLogging.callIdHeader")?.getString() ?: "X-Request-Id"
     install(CallLogging) {
         level = org.slf4j.event.Level.INFO
+        callIdMdc(callIdHeader)
     }
+    install(CallId) {
```

Comment:
Started by implementing our own interceptor implementation, but discovered that ktor already does this for us. We just have to register our generation implementation, and define the header name we want to use for trace requests.